bibtex-completion.el

```diff
@@ -1140,11 +1140,14 @@ The format depends on
 Return DEFAULT if FIELD is not present in ENTRY."
   ;; Virtual fields:
   (pcase field
-    ("author-or-editor"
-     (if-let ((value (bibtex-completion-get-value "author" entry)))
-         (bibtex-completion-apa-format-authors value)
-       (bibtex-completion-apa-format-editors
-        (bibtex-completion-get-value "editor" entry))))
+    ((or "author-or-editor"
+         "author-or-editor-abbrev")
+     (let ((value (cl-some (lambda (field)
+                             (bibtex-completion-get-value field entry))
+                           (list "author" "editor"))))
+       (if (string-match-p "abbrev$" field)
+           (bibtex-completion-apa-format-authors value)
+         (bibtex-completion-apa-format-authors-abbrev value))))
```

Comment:
Note that we're using the `format-authors` and `format-authors-abbrev`, since, if there aren't any author, we'd like to treat editors just as if they were themselves authors.

Reply:
This is neat but note that APA mandates that authors and editors are formatted differently (last first vs fist last). Perhaps the logic behind this is, that the order disambiguates authors from editors so that you can't get them confused.

Reply:
So, we could keep the general idea of having a new `author-or-editor-abbrev` expansion, but just adjust the order of names?  I could fix that.

Reply:
Yes. We could add virtual field "editor-abbrev" and then recurse when "author-or-editor-abbrev" is requested.

Reply:
Right, I'll need to modify `bibtex-completion-apa-format-authors-abbrev` to do our bidding, then.  I'm a bit short on time right now, but I'll get to it within the fortnight.

Reply:
I'd create a separate function `bibtex-completion-apa-firmat-editors-abbrev` for that purpose.  Otherwise the naming becomes misleading. But these two functions can of course share common code.

Reply:
Oh yeah, sorry, I meant to say 'modify _and_ create a new function', but we're on the same page.

Reply:
Prototyped in fb5b30195aa6540b1ac78cd4922fc1e165c6e375.